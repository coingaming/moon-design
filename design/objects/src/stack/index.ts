import styled from '@emotion/styled';
import { spacing, SpacingSize } from '@heathmont/sportsbet-utils';

export type StackProps = {
  space?: SpacingSize;
};

export const Stack = styled.div<StackProps>(({ space }) => ({
  '& > * + *': {
    marginTop: spacing(space),
  },
}));
