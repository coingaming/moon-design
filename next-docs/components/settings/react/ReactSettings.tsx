import { useState } from 'react';
import {
  LogoSportsbetShort,
  LogoBitcasinoShort,
  LogoMoonDesignShort,
  LogoBetaddaShort,
} from '@heathmont/moon-assets';
import useFeatureFlags from '../../../utils/useFeatureFlags';
import { useDocsTheme, DocsBrands } from '../../themes/DocsThemeProvider';
import useThemeTW from '../../themes/useThemesTW';
import ReactElixirSwitcher from '../common/ReactElixirSwitcher';
import SettingsButton from '../common/SettingsButton';
import DarkLightModeSwitcher from './DarkLightModeSwitch';
import RTLModeSwitch from './RTLModeSwitch';

type LogosKeys = keyof typeof Logos;

interface Props {
  toggleReactAndElixir: () => void;
  isReactEnabled: boolean;
}

const Logos = {
  sportsbet: <LogoSportsbetShort fontSize="1rem" />,
  bitcasino: <LogoBitcasinoShort fontSize="1rem" />,
  moonDesign: <LogoMoonDesignShort fontSize="1rem" />,
  betadda: <LogoBetaddaShort fontSize="1rem" />,
};

const ReactSettings: React.FC<Props> = ({
  toggleReactAndElixir,
  isReactEnabled,
}) => {
  const [isOpened, setIsOpened] = useState(false);
  const toggle = () => setIsOpened(!isOpened);

  const { toggleColorScheme, setBrand, themeKeys } = useDocsTheme();

  const [isRtlEnabled, setIsRtl] = useState(false);

  const { isFeatureElixirEnabled } = useFeatureFlags();

  const toggleRtl = () => {
    const htmlTag = document && document?.getElementsByTagName('html')[0];
    setIsRtl(!isRtlEnabled);
    const dirAttr = htmlTag.getAttribute('dir');

    if (!dirAttr) {
      htmlTag.setAttribute('dir', 'rtl');
    } else {
      htmlTag.removeAttribute('dir');
    }
  };

  const filteredThemeKeys = themeKeys.filter((themeKey: string) =>
    ['sportsbet', 'bitcasino', 'moonDesign', 'betadda'].includes(themeKey)
  );

  const {
    setBrand: setBrandTW,
    toggleMode: toggleModeTW,
    getMode,
  } = useThemeTW();

  const switchBrandHandler = (themeKey: LogosKeys) => {
    setBrand(themeKey as DocsBrands);
    setBrandTW(themeKey);
  };

  const switchModeHandler = () => {
    toggleColorScheme();
    toggleModeTW();
  };
  return (
    <>
      <SettingsButton onClick={toggle} isRtlEnabled={isRtlEnabled} />
      {isOpened && (
        <>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-16 z-50`}
          >
            {filteredThemeKeys.map((themeKey: string) => {
              const Logo = Logos[themeKey as LogosKeys] || Logos.moonDesign;
              return (
                <button
                  key={themeKey}
                  type="button"
                  onClick={() => switchBrandHandler(themeKey as LogosKeys)}
                  className={`${
                    isRtlEnabled ? `mr-4` : `ml-4`
                  } p-2 rounded-full text-black bg-white hover:bg-gray-200 inline-flex items-center justify-center shadow-md`}
                >
                  {Logo}
                </button>
              );
            })}
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-28 z-50`}
          >
            <DarkLightModeSwitcher
              toggle={switchModeHandler}
              isEnabled={getMode() === 'dark'}
              isRtlEnabled={isRtlEnabled}
            />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-40 z-50`}
          >
            <RTLModeSwitch toggle={toggleRtl} isEnabled={isRtlEnabled} />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-52 z-50`}
          >
            {isFeatureElixirEnabled ? (
              <ReactElixirSwitcher
                toggle={toggleReactAndElixir}
                isReactEnabled={isReactEnabled}
              />
            ) : null}
          </div>
        </>
      )}
    </>
  );
};

export default ReactSettings;
