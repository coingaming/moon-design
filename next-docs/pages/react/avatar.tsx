import React from 'react';

import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageAvatar() {
  return (
    <>
      <ThemeProvider theme={sportsbetDark}>
        <h1>Avatar</h1>
      </ThemeProvider>
    </>
  );
}
