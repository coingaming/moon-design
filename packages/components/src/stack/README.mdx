---
name: Stack
menu: Components
route: /components/stack
---

# Stack

> Based on the "Stack" component from [every-layout.dev](https://every-layout.dev/layouts/stack/).

Components are designed to fit any container, with the spacing between elements
set by the consumer to fit specific use-cases.

The `Stack` component is designed to meet the majority of these use-cases,
setting consistent **vertical rhythm** to its **immediate children**. Thanks to
its low specificity, margins can be overriden or adjusted for individual child
elements by creating a new `class` selector.

```jsx react-live
() => {
  const Example = styled.div(({ theme: { borderStyle, color, space } }) => ({
    height: rem(space.large),
    borderStyle,
    borderWidth: 3,
    borderColor: color.trunks[100],
  }));

  return (
    <Stack>
      <Example />
      <Example />
      <Stack>
        <Example />
        <Example />
      </Stack>
      <Example />
    </Stack>
  );
};
```

The immediate children don't _need_ to be components; `Stack` is especially
useful for long-form content.

```jsx react-live
<Stack>
  <Heading as="h1" size="bravo">
    Lorem Ipsum Dolor Sit Amet.
  </Heading>
  <p>
    Consectetur adipiscing elit. Nullam gravida ex tempor, ornare sapien a,
    sodales velit. Curabitur ac ultrices massa, eu scelerisque mauris.
  </p>
  <p>
    Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et
    vestibulum felis venenatis pellentesque. Donec malesuada diam nec magna
    maximus semper. In iaculis nunc erat, in dignissim tortor facilisis id.
    Nulla sit amet auctor nisi, eu porttitor tortor. Suspendisse congue porta
    magna, vel euismod eros tempus a. Vestibulum consequat consequat vulputate.
    Vivamus fringilla malesuada purus in consectetur.
  </p>
  <p>
    Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla.
    Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
  </p>
</Stack>
```

Alternatively, "Stack" styles can also be added to an existing component via the
`stack()` function:

```jsx
import { stack } from '@heathmont/sportsbet-components';

const Example = styled.div({
  ...stack('1.25rem'),
});
```

## Space

`Stack` uses the [`"default"`](/themes/tokens#space) spacing token to set rhythm, but
accepts other size values for individual use-cases.

```jsx react-live
() => {
  const Example = styled.div(({ theme: { borderStyle, color, space } }) => ({
    height: rem(space.large),
    borderStyle,
    borderWidth: 3,
    borderColor: color.trunks[100],
  }));

  return (
    <Stack space={spacing('small')}>
      <Example />
      <Example />
      <Example />
      <Example />
    </Stack>
  );
};
```

## Nesting

Only the **immediate** children will be affected by `Stack`, anything further
nested will lose spacing. Further `Stack`s can be nested to resolve this issue,
or for other spacing needs.

```jsx react-live
() => {
  const Example = styled.div(
    ({ theme: { border, color, space }, padding }) => ({
      border,
      borderColor: color.trunks[100],
      padding: padding && rem(space[padding]),
    })
  );

  const NestedStack = styled(Stack)(({ theme: { border, color, space } }) => ({
    border,
    borderColor: color.trunks[100],
    padding: rem(space.default),
  }));

  return (
    <Stack>
      <p>Stacks can be nested too.</p>
      <Example padding="default">
        <strong>Content unaffected by Stack.</strong>
        <p>
          Elements that are not direct children of the Stack, such as this one,
          will lose their spacing.
        </p>
      </Example>

      <NestedStack>
        <strong>New Stack defined for nested elements.</strong>
        <p>Immediate children of this Stack will benefit from spacing.</p>
        <p>Such as this element.</p>
      </NestedStack>

      <Example padding="medium" />

      <p>
        Further immediate children to the initial Stack will inherit spacing.
      </p>
    </Stack>
  );
};
```
