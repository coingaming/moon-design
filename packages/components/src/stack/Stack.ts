import styled, { CSSObject } from 'styled-components';
import { themed } from '@heathmont/sportsbet-utils';
import { SpaceProps } from '@heathmont/sportsbet-themes';

import stackMixin from '../stackMixin/stackMixin';

type StackSpace = CSSObject['marginTop'] | CSSObject['gridGap'];

export type StackProps = {
  /**
   * Excepts a string value, number or a theme key.
   *
   * Theme keys are returned as a `rem`-based unit.
   *
   * e.g. `<Stack space="large" />` returns `space.large` as a rem unit.
   */
  space?: SpaceProps | StackSpace;
};

const Stack = styled.div<StackProps>(({ theme, space = 'default' }) =>
  stackMixin(themed('space', space)(theme))
);

export default Stack;
