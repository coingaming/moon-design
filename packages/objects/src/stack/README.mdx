---
name: Stack
menu: Objects
route: /objects/stack
---

# Stack

> Based on the "Stack" component from [every-layout.dev](https://every-layout.dev/layouts/stack/).

Components are designed to fit any container, with the spacing between elements
set by the consumer to fit specific use-cases.

The `Stack` component is designed to meet the majority of these use-cases,
setting consistent **vertical rhythm** to its **immediate children**. Thanks to
its low specificity, margins can be overriden or adjusted for individual child
elements by creating a new `class` selector.

```jsx react-live
() => {
  const Example = styled.div(({ css }) => css);

  return (
    <Stack>
      <Example
        css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
      />
      <Example
        css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
      />
      <Stack
        css={{ padding: '1rem', border: `3px solid ${colors.neutral[20]}` }}
      >
        <Example
          css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
        />
        <Example
          css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
        />
      </Stack>
      <Example
        css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
      />
    </Stack>
  );
};
```

The immediate children don't _need_ to be components; `Stack` is especially
useful for long-form content.

```jsx react-live
<Stack>
  <Heading as="h1" size="bravo">
    Lorem Ipsum Dolor Sit Amet.
  </Heading>
  <p>
    Consectetur adipiscing elit. Nullam gravida ex tempor, ornare sapien a,
    sodales velit. Curabitur ac ultrices massa, eu scelerisque mauris.
  </p>
  <p>
    Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et
    vestibulum felis venenatis pellentesque. Donec malesuada diam nec magna
    maximus semper. In iaculis nunc erat, in dignissim tortor facilisis id.
    Nulla sit amet auctor nisi, eu porttitor tortor. Suspendisse congue porta
    magna, vel euismod eros tempus a. Vestibulum consequat consequat vulputate.
    Vivamus fringilla malesuada purus in consectetur.
  </p>
  <p>
    Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla.
    Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
  </p>
</Stack>
```

Alternatively, "Stack" styles can also be added to an existing component via the
`stack()` function:

```jsx
import { stack } from '@heathmont/sportsbet-objects';

const Example = styled.div({
  ...stack('1.25rem'),
});
```

## Space

`Stack` uses the [`"default"`](/tokens#spacing) spacing token to set rhythm, but
accepts other size values (such as [spacing utils](/utils/spacing)) for
individual use-cases.

```jsx react-live
() => {
  const Example = styled.div(({ css }) => css);

  return (
    <Stack space={spacing('small')}>
      <Example
        css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
      />
      <Example
        css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
      />
      <Example
        css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
      />
      <Example
        css={{ height: '2rem', border: `3px solid ${colors.neutral[20]}` }}
      />
    </Stack>
  );
};
```

## Nesting

Only the **immediate** children will be affected by `Stack`, anything further
nested will lose spacing. Further `Stack`s can be nested to resolve this issue,
or for other spacing needs.

```jsx react-live
() => {
  const Example = styled.div(({ css }) => css);

  const NestedStack = styled(Stack)({
    padding: '1rem',
    border: `1px solid ${colors.neutral[20]}`,
  });

  return (
    <Stack>
      <p>Stacks can be nested too.</p>
      <Example
        css={{ padding: '1rem', border: `1px solid ${colors.neutral[20]}` }}
      >
        <strong>Content unaffected by Stack.</strong>
        <p>
          Elements that are not direct children of the Stack, such as this one,
          will lose their spacing.
        </p>
      </Example>

      <NestedStack>
        <strong>New Stack defined for nested elements.</strong>
        <p>Immediate children of this Stack will benefit from spacing.</p>
        <p>Such as this element.</p>
      </NestedStack>

      <Example
        css={{ height: '2rem', border: `1px solid ${colors.neutral[20]}` }}
      />

      <p>
        Further immediate children to the initial Stack will inherit spacing.
      </p>
    </Stack>
  );
};
```

## Horizontal

By default, margins are set on the `vertical` axis but this can switched to
`horizontal` for specific use-cases (such as spacing in [`flex`-based](https://developer.mozilla.org/en-US/docs/Web/CSS/flex) design).

```jsx react-live
() => {
  const Example = styled.div(({ css }) => css);

  const CustomStack = styled(Stack)({
    display: 'flex',
    minHeight: '2rem',
  });

  return (
    <CustomStack direction="horizontal">
      <Example
        css={{ width: '100%', border: `3px solid ${colors.neutral[20]}` }}
      />
      <Example
        css={{ width: '100%', border: `3px solid ${colors.neutral[20]}` }}
      />
      <Example
        css={{ width: '100%', border: `3px solid ${colors.neutral[20]}` }}
      />
    </CustomStack>
  );
};
```
