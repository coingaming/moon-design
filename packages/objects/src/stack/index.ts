import styled from '@emotion/styled';
import { spacing } from '@heathmont/sportsbet-utils';

const disableSSRWarning = (selector: string) =>
  `${selector} /* emotion-disable-server-rendering-unsafe-selector-warning-please-do-not-use-this-the-warning-exists-for-a-reason */`;

type StackSpace = string | number;

export type StackProps = {
  space?: StackSpace;
};

export const stack = (space: StackSpace) => ({
  /**
   * If a browser supports the `grid-gap` property, let's use it.
   * Otherwise, fallback to the lobotomised owl selector to style children.
   */
  '@supports (grid-gap: 0)': {
    display: 'grid',
    gridGap: space,
  },
  '@supports not (grid-gap: 0)': {
    '& > * + *': {
      marginTop: space,
    },
    /**
     * With Emotion's 'out of the box' SSR set-up, <style/> tags are injected
     * inline above each element with styles on first load.
     *
     * This means that if the first-child of <Stack/> has styles, the first-child
     * of <Stack/> will  be a <style/> tag, then the actual first-child - in this
     * case the first-child becomes the second and inherits the 'marginTop'.
     *
     * For now, if the first-child is a <style/> tag on load, let's make sure
     * the adjacent element to it (our actual first-child), has an enforced
     * 'marginTop' of 0.
     * https://github.com/emotion-js/emotion/issues/1178
     */
    [disableSSRWarning('& > style:first-child + *')]: {
      marginTop: 0,
    },
  },
});

export const Stack = styled.div<StackProps>(({ space = spacing('default') }) =>
  stack(space)
);
