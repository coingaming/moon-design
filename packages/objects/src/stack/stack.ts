import styled from 'styled-components';
import { spacing } from '@heathmont/sportsbet-utils';

type StackSpace = string | number;
type StackDirection = 'vertical' | 'horizontal';

export type StackProps = {
  space?: StackSpace;
  direction?: StackDirection;
};

export const stack = (
  space: StackSpace,
  direction: StackDirection = 'vertical'
) => {
  const margin = direction === 'vertical' ? 'marginTop' : 'marginLeft';

  const margins = {
    '& > * + *': {
      [margin]: space,
    },
    /**
     * With Emotion's 'out of the box' SSR set-up, <style/> tags are injected
     * inline above each element with styles on first load.
     *
     * This means that if the first-child of <Stack/> has styles, the first-child
     * of <Stack/> will  be a <style/> tag, then the actual first-child - in this
     * case the first-child becomes the second and inherits the 'marginTop'.
     *
     * For now, if the first-child is a <style/> tag on load, let's make sure
     * the adjacent element to it (our actual first-child), has an enforced
     * 'marginTop' of 0.
     * https://github.com/emotion-js/emotion/issues/1178
     */
    '& > style:first-child + *': {
      [margin]: 0,
    },
  };

  return direction === 'vertical'
    ? {
        /**
         * If a browser supports the `grid-gap` property, let's use it.
         * Otherwise, fallback to the lobotomised owl selector to style children.
         */
        '@supports (grid-gap: 0)': {
          display: 'grid',
          gridGap: space,
        },
        '@supports not (grid-gap: 0)': {
          ...margins,
        },
      }
    : margins;
};

export const Stack = styled.div<StackProps>(
  ({ space = spacing('default'), direction = 'vertical' }) =>
    stack(space, direction)
);
