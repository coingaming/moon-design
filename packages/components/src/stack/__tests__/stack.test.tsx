import * as React from 'react';
import { create } from 'react-test-renderer';
import 'jest-styled-components';

import { Stack } from '../stack';

const TestItems = () => (
  <>
    <h1>Lorem Ipsum Dolor Sit Amet.</h1>
    <p>
      Consectetur adipiscing elit. Nullam gravida ex tempor, ornare sapien a,
      sodales velit. Curabitur ac ultrices massa, eu scelerisque mauris.
    </p>
    <p>
      Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et
      vestibulum felis venenatis pellentesque.
    </p>
    <p>
      Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla.
      Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
    </p>
    <div>
      <p>
        Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et
        vestibulum felis venenatis pellentesque.
      </p>
      <p>
        Phasellus felis sem, convallis eget pretium vel, faucibus sit amet
        nulla. Fusce auctor nibh non libero sagittis, semper sagittis ipsum
        euismod.
      </p>
    </div>
  </>
);

describe('Stack', () => {
  test('renders with `default` spacing', () => {
    const stack = create(
      <Stack>
        <TestItems />
      </Stack>
    );

    expect(stack).toMatchSnapshot();
  });

  test('renders with adjusted spacing', () => {
    const stack = create(
      <Stack space={30}>
        <TestItems />
      </Stack>
    );

    expect(stack).toMatchSnapshot();
  });
});
