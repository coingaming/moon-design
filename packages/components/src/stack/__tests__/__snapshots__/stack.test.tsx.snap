// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack renders with \`default\` spacing 1`] = `
.c0 > li {
  list-style-type: none;
}

.c0 > li:before {
  position: absolute;
  content: "\\200B";
}

@supports (grid-gap:0) {
  .c0 {
    display: grid;
    grid-gap: 1rem;
  }
}

@supports not (grid-gap:0) {
  .c0 > * + * {
    margin-top: 1rem;
  }
}

<div
  className="c0"
>
  <h1>
    Lorem Ipsum Dolor Sit Amet.
  </h1>
  <p>
    Consectetur adipiscing elit. Nullam gravida ex tempor, ornare sapien a, sodales velit. Curabitur ac ultrices massa, eu scelerisque mauris.
  </p>
  <p>
    Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et vestibulum felis venenatis pellentesque.
  </p>
  <p>
    Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla. Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
  </p>
  <div>
    <p>
      Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et vestibulum felis venenatis pellentesque.
    </p>
    <p>
      Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla. Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
    </p>
  </div>
</div>
`;

exports[`Stack renders with adjusted spacing 1`] = `
.c0 > li {
  list-style-type: none;
}

.c0 > li:before {
  position: absolute;
  content: "\\200B";
}

@supports (grid-gap:0) {
  .c0 {
    display: grid;
    grid-gap: 30px;
  }
}

@supports not (grid-gap:0) {
  .c0 > * + * {
    margin-top: 30px;
  }
}

<div
  className="c0"
>
  <h1>
    Lorem Ipsum Dolor Sit Amet.
  </h1>
  <p>
    Consectetur adipiscing elit. Nullam gravida ex tempor, ornare sapien a, sodales velit. Curabitur ac ultrices massa, eu scelerisque mauris.
  </p>
  <p>
    Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et vestibulum felis venenatis pellentesque.
  </p>
  <p>
    Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla. Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
  </p>
  <div>
    <p>
      Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et vestibulum felis venenatis pellentesque.
    </p>
    <p>
      Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla. Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
    </p>
  </div>
</div>
`;

exports[`Stack renders with theme key 1`] = `
.c0 > li {
  list-style-type: none;
}

.c0 > li:before {
  position: absolute;
  content: "\\200B";
}

@supports (grid-gap:0) {
  .c0 {
    display: grid;
    grid-gap: 2rem;
  }
}

@supports not (grid-gap:0) {
  .c0 > * + * {
    margin-top: 2rem;
  }
}

<div
  className="c0"
>
  <h1>
    Lorem Ipsum Dolor Sit Amet.
  </h1>
  <p>
    Consectetur adipiscing elit. Nullam gravida ex tempor, ornare sapien a, sodales velit. Curabitur ac ultrices massa, eu scelerisque mauris.
  </p>
  <p>
    Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et vestibulum felis venenatis pellentesque.
  </p>
  <p>
    Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla. Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
  </p>
  <div>
    <p>
      Donec maximus purus at tempor elementum. Donec commodo aliquam metus, et vestibulum felis venenatis pellentesque.
    </p>
    <p>
      Phasellus felis sem, convallis eget pretium vel, faucibus sit amet nulla. Fusce auctor nibh non libero sagittis, semper sagittis ipsum euismod.
    </p>
  </div>
</div>
`;
